boards/itsybitsy_m0/src/pins.rs

```diff
@@ -0,0 +1,335 @@
+    pin d10 = a18,
+    pin d11 = a16,
+    pin d12 = a19,
+    /// Digital pin number 13, which is also attached to
+    /// the red LED.  PWM capable.
+    pin d13 = a17,
+
+    /// The SPI MOSI
+    pin mosi = b10,
+    /// The SPI MISO
+    pin miso = a12,
+    /// The SPI SCK
+    pin sck = b11,
+
+    /// The MOSI pin attached to the on-board SPI flash
+    pin flash_mosi = b22,
+    /// The MISO pin attached to the on-board SPI flash
+    pin flash_miso = b3,
+    /// The SCK pin attached to the on-board SPI flash
+    pin flash_sck = b23,
+    /// The CS pin attached to the on-board SPI flash
+    pin flash_cs = a27,
+
+    /// The I2C clock
+    pin scl = a23,
+    /// The I2C data line
+    pin sda = a22,
+
+    /// The DotStar clock
+    pin dotstar_ci = a0,
+    /// The DotStar data line
+    pin dotstar_di = a1,
+    /// Not connected, but usable as the MISO when addressing
+    /// the dotstar over SPI.
+    pin dotstar_nc = b0,
+
+    pin swdio = a31,
+    pin swdclk = a30,
+
+    /// The USB D- pad
+    pin usb_dm = a24,
+    /// The USB D+ pad
+    pin usb_dp = a25,
+);
+
+impl Pins {
+    /// Split the device pins into subsets
+    pub fn split(self) -> Sets {
+        let analog = Analog {
+            a0: self.a0,
+            a1: self.a1,
+            a2: self.a2,
+            a3: self.a3,
+            a4: self.a4,
+            a5: self.a5,
+        };
+
+        let dotstar = Dotstar {
+            ci: self.dotstar_ci,
+            di: self.dotstar_di,
+            nc: self.dotstar_nc,
+        };
+
+        let spi = SPI {
+            sck: self.sck,
+            mosi: self.mosi,
+            miso: self.miso,
+        };
+
+        let i2c = I2C {
+            sda: self.sda,
+            scl: self.scl,
+        };
+
+        let usb = USB {
+            dm: self.usb_dm,
+            dp: self.usb_dp,
+        };
+
+        let uart = UART {
+            rx: self.d0,
+            tx: self.d1,
+        };
+
+        Sets {
+            analog,
+            dotstar,
+            spi,
+            i2c,
+            usb,
+            uart,
+            port: self.port,
+        }
+    }
+}
+
+/// Sets of pins split apart by category
+pub struct Sets {
+    /// Analog pins
+    pub analog: Analog,
+
+    /// Dotstar (RGB LED) pins
+    pub dotstar: Dotstar,
+
+    /// SPI (external pinout) pins
+    pub spi: SPI,
+
+    /// I2C (external pinout) pins
+    pub i2c: I2C,
+
+    /// USB pins
+    pub usb: USB,
+
+    /// UART (external pinout) pins
+    pub uart: UART,
+
+    /// Port
+    pub port: Port,
+}
+
+/// Analog pins
+pub struct Analog {
+    pub a0: Pa2<Input<Floating>>,
+    pub a1: Pb8<Input<Floating>>,
+    pub a2: Pb9<Input<Floating>>,
+    pub a3: Pa4<Input<Floating>>,
+    pub a4: Pa5<Input<Floating>>,
+    pub a5: Pb2<Input<Floating>>,
+}
+
+/// Dotstar pins
+pub struct Dotstar {
+    pub ci: Pa0<Input<Floating>>,
+    pub di: Pa1<Input<Floating>>,
+    pub nc: Pb0<Input<Floating>>,
+}
+
+impl Dotstar {
```

Comment:
We dont test well for this on ci, but you're going to need to cfg unproven some of this so it builds even without unproven
Use `cargo build --no-default-features --features "rt,atsamd-hal/samd21g18a"` locally until that can build successfully